### PR TITLE
fix thumbnail allow_upscale option

### LINKED ISSUE #560: allow_upscale on thumbnail filter cant work due to imagine
I tried to use the allow_upscale option on a thumbnail filter but the file did not upscale. So i debugged and found this in https://github.com/avalanche123/Imagine/blob/v0.6.1/lib/Imagine/Image/AbstractImage.php:

``` php
// if target width is larger than image width
// AND target height is longer than image height
if ($size->contains($imageSize)) {
    return $thumbnail;
}
```

I do not see any short workaround on that as far as this is code form imagine.

### PR DESCRIPTION
**Pull request by yplam**
Fix thumbnail allow_upscale option

For smaller image, resize first before creating thumbnail.

Resolve #560 and #428

### CODE REVIEW

**Comment by makasim** [comment]
Could you please add a test?

File: Imagine/Filter/Loader/ThumbnailFilterLoader.php [hunk 1]
```diff
@@ -42,6 +43,12 @@ public function load(ImageInterface $image, array $options = array())
         if (($origWidth > $width || $origHeight > $height)
             || (!empty($options['allow_upscale']) && ($origWidth !== $width || $origHeight !== $height))
         ) {
+            if(($origWidth < $width || $origHeight < $height)){
+                // resize first
+                $ratio = max($width / $origWidth, $height / $origHeight);
+                $resizeFilter = new Resize(new Box(round($origWidth * $ratio), round($origHeight * $ratio)));
+                $image = $resizeFilter->apply($image);
+            }
```
Comment (trsteel88) [inline Imagine/Filter/Loader/ThumbnailFilterLoader.php]: This is duplicate code. Why not just do:

```php
$upscale = new [UpscaleFilterLoader()](https://github.com/liip/LiipImagineBundle/blob/master/Imagine/Filter/Loader/UpscaleFilterLoader.php);
$image = $upscale->load($image, array('min' => array($width, $height));

**Comment by lsmith77** [comment]
ping

**Comment by lsmith77** [comment]
please rebase on master to get the test fixed

**Comment by JulienItard** [comment]
Need it !

**Comment by lsmith77** [comment]
ping

File: Imagine/Filter/Loader/ThumbnailFilterLoader.php (hunk 1 above)
Comment (lsmith77) [inline Imagine/Filter/Loader/ThumbnailFilterLoader.php]: cab you bake this into a PR of your own?

**Pull request closed without merging**